zeebe/broker/src/main/java/io/camunda/zeebe/broker/transport/ErrorResponseWriter.java

```diff
@@ -125,7 +126,15 @@ public ErrorResponseWriter processNotFound(final String processIdentifier) {
         .errorMessage(String.format(PROCESS_NOT_FOUND_FORMAT, processIdentifier));
   }
 
-  public ErrorResponseWriter mapWriteError(final int partitionId, final WriteFailure error) {
+  public ErrorResponseWriter mapWriteError(
+      final int partitionId, final WriteFailure error, final boolean processingPaused) {
+    if (processingPaused && error == FULL) {
```

Comment:
:x: The buffer might not be full when processing is paused. A better solution would be to reject the command before even attempting to write the command in `CommandApiRequestHandler#handleExecuteCommandRequest` if the partition has paused processing.